flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitGenerator.java

```diff
@@ -37,9 +40,20 @@ private FlinkSplitGenerator() {
   static FlinkInputSplit[] createInputSplits(Table table, ScanContext context) {
     List<CombinedScanTask> tasks = tasks(table, context);
     FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
-    for (int i = 0; i < tasks.size(); i++) {
-      splits[i] = new FlinkInputSplit(i, tasks.get(i));
-    }
+    boolean exposeLocality = context.exposeLocality();
+
+    Tasks.range(tasks.size())
+        .stopOnFailure()
+        .executeWith(exposeLocality ? ThreadPools.getWorkerPool() : null)
+        .run(index -> {
+          CombinedScanTask task = tasks.get(index);
+          String[] hosts = new String[0];
```

Comment:
should we just initialize it to `null` and remove the `? : ` part later?

Reply:
Such is the case.